src/olympia/github/utils.py

```diff
@@ -0,0 +1,149 @@
+import os
+import uuid
+import zipfile
+
+import commonware.log
+import requests
+from django import forms
+from django.conf import settings
+from django.core.files.storage import default_storage as storage
+from django_statsd.clients import statsd
+
+from olympia.amo.helpers import user_media_path
+from olympia.files.utils import SafeUnzip
+
+log = commonware.log.getLogger('z.github')
+
+
+class GithubCallback(object):
+
+    def __init__(self, data):
+        if data['type'] != 'github':
+            raise ValueError('Not a github callback.')
+        self.data = data
+
+    def get(self):
+        log.info('Getting zip from github: {}'.format(self.data['zip_url']))
+        with statsd.timer('github.zip'):
+            res = requests.get(self.data['zip_url'])
+            res.raise_for_status()
+        return res
+
+    def post(self, url, data):
+        msg = data.get('state', 'comment')
+        log.info('Setting github to: {} at: {}'.format(msg, url))
+        with statsd.timer('github.{}'.format(msg)):
+            data['context'] = 'addons/linter'
```

Comment:
Should this contain some information about Mozilla? Also, I see that GitHub doesn't do it but let's maybe try to send the olympia release with it? That'll help debug weird "it failed but it doesn't locally" problems with people not updating their packages.